components/sessionModals/SessionTimeTracker.js

```diff
@@ -0,0 +1,92 @@
+import Modal from 'react-modal'
+import { useState, useEffect } from 'react'
+import CountDown from './CountDown'
+import Router from 'next/router'
+
+const SessionTimeTracker = ({ popupStaySignedIn, aaPrefix }) => {
+  const initialExpires = () => {
+    const currentDate = new Date()
+    return {
+      warning: new Date(currentDate.getTime() + 0.5 * 60 * 1000),
+      logout: new Date(currentDate.getTime() + 1 * 60 * 1000),
+    }
+  }
+  const [expires, setExpires] = useState({ ...initialExpires(), active: true })
+
+  const [demoModalBody, setDemoModalBody] = useState(null)
+
+  function demoContent(content) {
+    setDemoModalBody(content)
+  }
+
+  function closeDemoModal() {
+    setDemoModalBody(null)
+  }
+
+  function closeModal() {
+    setOpenModalWithLink({ isOpen: false, activeLink: '/' })
+  }
+
+  useEffect(() => {
+    const updateModalTimer = (ev) => {
+      if (!demoModalBody) {
+        setExpires((prev) => {
+          return {
+            ...prev,
+            ...initialExpires(),
+          }
+        })
+      }
+    }
+    window.addEventListener('mousemove', updateModalTimer)
+    window.addEventListener('scroll', updateModalTimer)
+    return () => {
+      window.removeEventListener('mousemove', updateModalTimer)
+      window.removeEventListener('scroll', updateModalTimer)
+    }
```

Comment:
While this is okay, I feel there's a lot more events we can track to determine if a user is idle. See the default props for `react-idle-timer` here[](https://idletimer.dev/docs/api/props#events) as an example.

Reply:
I've implemented react-idle-timer, and will now focus on the modal comment you'd made.